src/main/java/seedu/address/ui/CalendarPanel.java

```diff
@@ -0,0 +1,341 @@
+        }
+        return result;
+    }
+
+    /**
+     * Generates array containing days from next month.
+     * @param currMonth Current selected month
+     * @param tailGap Number of days needed from next month
+     * @return String array containing days from next month.
+     */
+    private String[] nextMonth(int currMonth, int tailGap) {
+        Calendar c = Calendar.getInstance();
+        c.set(Calendar.YEAR, year);
+        c.set(Calendar.MONTH, currMonth);
+        c.add(Calendar.MONTH, 1);
+        int y = c.get(Calendar.YEAR);
+        int m = c.get(Calendar.MONTH);
+
+        GregorianCalendar gc = new GregorianCalendar(y, m, 1);
+        int daysInMonth = DAYS_IN_MONTH[m];
+        if (gc.isLeapYear(gc.get(Calendar.YEAR)) && m == 1) {
+            daysInMonth++;
+        }
+
+        StringBuilder sb = new StringBuilder();
+        for (int i = 1; i <= daysInMonth; i++) {
+            sb.append(i).append(" ");
+        }
+        String[] allDays = sb.toString().split("\\s");
+        String[] neededDays = Arrays.copyOfRange(allDays, 0, tailGap);
+        return neededDays;
+    }
+
+    /**
+     * Combines 3 string arrays together, with array {@code a}'s elements coming in front, array
+     * {@code b}'s elements coming in the middle and array {@code c}'s elements coming at the back.
+     * @param a String array whose elements are to be in front
+     * @param b String array whose elements are to be in the middle
+     * @param c String array whose elements are to be behind
+     * @return String[] Combined String array.
+     */
+    private String[] combine(String[] a, String[] b, String[] c) {
+        int length = a.length + b.length + c.length;
+        String[] result = new String[length];
+        System.arraycopy(a, 0, result, 0, a.length);
+        System.arraycopy(b, 0, result, a.length, b.length);
+        System.arraycopy(c, 0, result, a.length + b.length, c.length);
+        return result;
+    }
+
+    /**
+     * Creates a label to be placed inside each grid of the grid pane in the FXML.
+     * @param numBefore Number of days from month before
+     * @param numCurr Number of days from selected month
+     * @param days String array containing all 42 days to be used to fill up the 7 * 6 grid pane
+     */
+    private void fillUpGrid(int numBefore, int numCurr, String[] days) {
+        int beforeCount = numBefore;
+        int currentCount = numCurr;
+        int counter = 0;
+        for (int i = 0; i < 6; i++) {
+            int row = i;
+            for (int j = 0; j < 7; j++) {
+                int column = j;
+                HBox h = new HBox();
+                Label l = new Label();
+                l.setText(days[counter]);
+                l.setPadding(new Insets(5, 0, 0, 10));
+                l.setFont(new Font("System", 11));
+                l.setAlignment(Pos.TOP_LEFT);
+                if (beforeCount > 0) {
+                    l.setTextFill(Paint.valueOf("#999999"));
```

Comment:
Great attention to detail in making the colour of the days of the previous month lighter.

Reply:
thanks!